apps/next/app/api/deploy/[tag]/status/route.ts

```diff
@@ -46,12 +55,33 @@ const mockGetDeploymentStatus = async (deploymentTag: string): Promise<Deploymen
   });
 };
 
-async function* makeIterator() {
-  yield encoder.encode('Deployment in progress...')
-  await sleep(5000)
-  yield encoder.encode('Cleaning deployment files...')
-  await sleep(5000)
-  yield encoder.encode('Deployed...')
+async function* makeIterator(buildTag: string) {
+  yield encoder.encode('Planning build...');
+  const { stdout: stdoutJson } = await exec('nixpacks plan ./temp');
+  const plan = JSON.parse(stdoutJson) as NixpackPlan;
+  // Filtering out 'npm-9_x' because for some reason it's not listed on:
+  // https://search.nixos.org/packages
+  plan.phases.setup.nixPkgs =
+    plan.phases.setup.nixPkgs.filter(nixPackage => nixPackage !== 'npm-9_x');
+  writeFileSync('./temp/build-plan.json', JSON.stringify(plan, null, 2), "utf8");
+  await sleep(1000);
+
+  yield encoder.encode('Generating Dockerfile...');
+  await exec(`nixpacks build ./temp --name ${buildTag} --config build-plan.json --out temp`);
```

Comment:
i think it'd be cool for the output of this command to pipe back to the user. I think this might involve switching to `spawn` or the async version of exec that gives you a stdout and stderr stream you can attach listeners to

Reply:
Ok. I'll take a look at this tomorrow.